itinerary_generator/generate_itinerary.py

```
@@ -0,0 +1,113 @@
+                pdf_path = convert_to_pdf(html_path, pdf_path, gotenberg_endpoint)
+                return html_path, pdf_path
+            except Exception as e:
+                import logging
+                logging.warning(f"PDF conversion failed: {e}. HTML output is still available.")
+                return html_path, None
+        
+        return html_path, None
+        
+    except FileNotFoundError as e:
+        # Re-raise with more context
+        raise FileNotFoundError(f"Failed to generate itinerary: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)